src/main/webapp/app/exercises/programming/manage/grading/testcase-analysis/testcase-analysis.component.ts

```diff
@@ -0,0 +1,118 @@
+        this.tasks = this.programmingExerciseTaskService.updateTasks();
+    }
+
+    // Iterate over all participations, get feedback details for each result, and filter them for negative feedback
+    loadFeedbacks(participations: Participation[]): void {
+        const MAX_CONCURRENT_REQUESTS = 5; // Maximum number of parallel requests
+
+        from(participations)
+            .pipe(
+                mergeMap((participation) => {
+                    return from(participation.results ?? []).pipe(
+                        mergeMap((result) => {
+                            return this.resultService.getFeedbackDetailsForResult(participation.id!, result).pipe(
+                                catchError(() => {
+                                    return of({ body: [] });
+                                }),
+                            );
+                        }, MAX_CONCURRENT_REQUESTS),
+                    );
+                }, MAX_CONCURRENT_REQUESTS),
+                toArray(),
+            )
+            .subscribe((responses) => {
+                const feedbackArray = responses.flatMap((response) => response.body ?? []);
+                const negativeFeedbackArray = feedbackArray.filter((feedback) => !feedback.positive); // Filter out positive feedback
+                this.saveFeedbacks(negativeFeedbackArray); // Save only negative feedback
+            });
+    }
+
+    // Iterate over all feedback and save them in the feedbacks array
+    // If a feedback with the corresponding testcase already exists in the list, then the count is incremented; otherwise, a new FeedbackDetail is added
+    saveFeedbacks(feedbackArray: Feedback[]): void {
+        const feedbackMap: Map<string, FeedbackDetail> = new Map();
+
+        feedbackArray.forEach((feedback) => {
+            const feedbackText = feedback.detailText ?? '';
+            const testcase = feedback.testCase?.testName ?? '';
+            const key = `${feedbackText}_${testcase}`;
+
+            if (feedbackMap.has(key)) {
+                const existingFeedback = feedbackMap.get(key);
+                if (existingFeedback) {
+                    existingFeedback.count += 1;
+                } // Increment count if feedback already exists
+            } else {
+                const task = this.findTaskIndexForTestCase(feedback.testCase); // Find the task index for the test case
+                feedbackMap.set(key, <FeedbackDetail>{
+                    count: 1,
+                    detailText: feedback.detailText ?? '',
+                    testcase: feedback.testCase?.testName,
+                    task: task,
+                });
+            }
+        });
+
+        // Convert map values to array and sort feedback by count in descending order
+        this.feedbacks = Array.from(feedbackMap.values()).sort((a, b) => b.count - a.count);
+    }
+
+    findTaskIndexForTestCase(testCase?: ProgrammingExerciseTestCase): number | undefined {
```

Comment:
Do we ever return `undefined` here?

```suggestion
    findTaskIndexForTestCase(testCase?: ProgrammingExerciseTestCase): number {
```